TROUBLESHOOTING.md

```diff
@@ -0,0 +1,192 @@
+# Troubleshooting Common Issues
+
+## Using this Document
+
+Search for the error message you're getting and see if there's a match, or simply skim the table of contents below for topics that seem relevant to the issue you're having. Each issue section has steps to work around or fix the particular issue, and have examples of common error messages.
+
+If you do not find the issue below, try searching the issue tracker itself for potential duplicates before opening a new issue.
+
+If you're reading this document because you noticed an issue with npm's web site, you may want to let the [web team](https://github.com/npm/www/issues) know.
+
+### Updating this Document
+
+If you think something should be added here, make a PR that includes the following:
+
+0. a summary
+0. one or more example errors
+0. steps to debug and fix
+0. links to at least one related issue from the tracker
+
+For more details of the content and formatting of these entries, refer to examples below.
+
+## Table of Contents
+
+* [Upgrading npm](#upgrading-npm)
+* [Proxies and Networking](#proxy-and-networking-issues)
+* [Cannot find module](#cannot-find-module)
+* [Shasum Check Fails](#shasum-check-fails)
+* [No Git](#no-git)
+
+## Upgrading npm
+
+Whenever you get npm errors, it's a good idea to first check your npm version and upgrade to latest whenever possible. We still see people running npm@1 (!) and in those cases, installing the latest version of npm usually solves the problem.
+
+You can check your npm version by running `npm -v`.
+
+### Steps to Fix
+* Upgrading on *nix (OSX, Linux, etc.)
+
+(You may need to prefix these commands with sudo, especially on Linux, or OS X if you installed Node using its default installer.)
+You can upgrade to the latest version of npm using:
+`npm install -g npm@latest`
+Or upgrade to the most recent LTS release:
+`npm install -g npm@lts`
+
+* Upgrading on Windows
+
+We have a [detailed guide](https://github.com/npm/npm/wiki/Troubleshooting#upgrading-on-windows) for upgrading on windows on our wiki.
+
+## Proxy and Networking Issues
+
+npm might not be able to connect to the registry for various reasons. Perhaps your machine is behind a firewall that needs to be opened, or you require a corporate proxy to access the npm registry. This issue can manifest in a wide number of different ways. Usually, strange network errors are simply this specific problem.
```

Comment:
"simply this specific problem" doesn't make sense

"Usually, strange network errors are the case of this specific problem" (if that's what you were trying to say here"